modules/library/main/src/main/java/org/geotools/data/util/NumericConverterFactory.java

```diff
@@ -325,6 +333,37 @@ static String toIntegral(String s) {
         }
     }
 
+    /**
+     * Clean the source handling possible edge cases.
```

Comment:
Should probably clarify what the edge cases are? My guess is that the issue happens when you have a target of type Long/Integer (integral in general) and the value is using scientific notation?

Reply:
At the moment only the case you mention justifies the presence of this extra code, and I tried to keep the change as clean as possible. Given that the computational cost is high related to the targeted cases. An alternative approach I was thinking could be make the conversion fails and then retry with a cleansed source.